vital/util/tests/test_span.cxx

```diff
@@ -0,0 +1,114 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief Test span template class.
+
+#include <vital/util/span.h>
+
+#include <gtest/gtest.h>
+
+#include <limits>
+
+using namespace kwiver::vital;
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+constexpr auto fnan = std::numeric_limits< float >::quiet_NaN();
+constexpr auto dnan = std::numeric_limits< double >::quiet_NaN();
+constexpr auto dinf = std::numeric_limits< double >::infinity();
+
+// ----------------------------------------------------------------------------
+TEST ( span, construct )
+{
+  {
+    auto const test_span = span< double >{ 5.0, 7.0 };
+    EXPECT_EQ( 5.0, test_span.lower() );
+    EXPECT_EQ( 7.0, test_span.upper() ); }
+
+  {
+    auto const test_span = span< int >{ 9, 2 };
+    EXPECT_EQ( 2, test_span.lower() );
+    EXPECT_EQ( 9, test_span.upper() );
+    EXPECT_EQ( span< int >( 2, 9 ), test_span );
```

Comment:
Is there a reason some places use `{}` initialization and others (e.g. here) use `()`?

Reply:
`EXPECT_EQ` is a macro, meaning it delimits arguments by commas. Matched parentheses escape the commas inside, but matched braces do not escape commas. Thank you, macros.

Reply:
Ah, right...